datafusion/core/src/execution/context.rs

```diff
@@ -3243,7 +3243,7 @@ mod tests {
             "SELECT cat, SUM(i) AS total FROM (
                     SELECT i, 'a' AS cat FROM catalog_a.schema_a.table_a
                     UNION ALL
-                    SELECT i, 'b' AS cat FROM catalog_b.schema_b.table_b
```

Comment:
Why we need to change it?

Reply:
to cover case when columns names in union inputs are not the same.

example:
SELECT 1 a UNION ALL SELECT 2;

Reply:
Got it. Thanks❤.

Reply:
I would recommend leaving this test alone and instead adding a new test, perhaps in https://github.com/apache/arrow-datafusion/blob/master/datafusion/core/tests/sql/union.rs using the reproducer from https://github.com/apache/arrow-datafusion/issues/2083

That way it is clear we haven't changed the coverage of this test and there is a more focused tests for just this particular change

Reply:
@gandronchik  -- What do you think about changing how this code is tested?